server/src/main/java/org/apache/iotdb/db/metadata/mnode/IMNode.java

```diff
@@ -88,6 +88,22 @@ public interface IMNode extends Serializable {
 
   boolean isMeasurement();
 
+  default MNodeType getMNodeType(Boolean isConfig) {
+    if (isEntity()) {
+      return MNodeType.DEVICE;
+    }
+    if (isStorageGroup()) {
+      return MNodeType.STORAGE_GROUP;
+    }
+    if (isMeasurement()) {
+      return MNodeType.MEASUREMENT;
+    }
+    if (isConfig) {
+      return MNodeType.SG_INTERNAL;
+    }
+    return MNodeType.INTERNAL;
+  }
```

Comment:
It's better to push down these if-else judgement to specific MNode class with method override.

Reply:
fixed~